nhost/service/manager.go

```diff
@@ -86,13 +95,162 @@ func (m *dockerComposeManager) SetGitBranch(gitBranch string) {
 	m.composeConfig = compose.NewConfig(m.nhostConfig, m.ports, m.env, gitBranch, m.projectName)
 }
 
+func (m *dockerComposeManager) checkPorts(ctx context.Context) error {
+	select {
+	case <-ctx.Done():
+		return nil
+	default:
+	}
+
+	type composeSvc struct {
+		Service    string `json:"Service"`
+		Publishers []struct {
+			PublishedPort uint32 `json:"PublishedPort"`
+		}
+	}
+
+	cmd, err := compose.WrapperCmd(ctx, []string{"ps", "--filter", "status=running", "--format", "json"}, m.composeConfig, nil)
```

Comment:
why do you need to list running containers?

Reply:
That's done to check whether the busy port is actually taken by a running service. In this case we skip all randomize actions